controllers/secret.go

```diff
@@ -55,13 +56,37 @@ func (r *OperatorPipelineReconciler) ensureGitHubAPISecret(ctx context.Context,
 	operatorPipeline, err := r.getPipeline(ctx, meta)
 	if err != nil {
 		log.Error(err, "unable to resolve github secret for %s/%s", meta.Namespace, meta.Name)
+		if err := r.updateStatusCondition(ctx, operatorPipeline, gitHubSecretAvailable, metav1.ConditionFalse, reconcileFailed,
+			err.Error()); err != nil {
+			return err
```

Comment:
I believe the logging is missing when that the status update fails. It has to be done either in here, or in `updateStatusCondition` method.

Reply:
I thought logging the condition update failure might be too much noise for a user in the controller. If an error happens in the update, the reconcile will be triggered again, and hopefully the condition update would happen again. If others agree we should log it, I'd prefer to go with a `WARN` for the level, since condition update is more of a `user friendly` feature and not a `must function` for the operator to work, type of thing.

Reply:
I agree with @acornett21 . I don't think this necessitates a log message.

Reply:
Going to resolve this comment, since this is more of a `user experience` nicety. Thanks everyone for the feedback.